### PR TITLE
ability to configure custom HttpClient

### PR DESCRIPTION
**Pull request by eli-darkly**
This is a copy of the user-contributed https://github.com/launchdarkly/dotnet-eventsource/pull/61 with some slight revisions as noted in PR comments.

### CODE REVIEW

File: src/LaunchDarkly.EventSource/Configuration.cs [hunk 1]
```diff
@@ -10,15 +10,15 @@ namespace LaunchDarkly.EventSource
     /// A class used 
     /// </summary>
     public sealed class Configuration
-    {
+    {
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/Configuration.cs:13]: There are some spurious changes like this in the diff, which I believe were due to the file having previously had inconsistent linefeeds from being edited on Mac vs. Windows. The linefeeds are now consistent, and you can see the PR more clearly by selecting "Hide whitespace changes" in GitHub.

As part of the 4.0 update I'll be fixing any other linefeed issues like this, and then using `.gitattributes` to ensure that linefeeds are translated correctly in future commits.

File: src/LaunchDarkly.EventSource/ConfigurationBuilder.cs [hunk 2]
```diff
@@ -1,135 +1,141 @@
...
+        /// <summary>
+        /// Sets the connection timeout value used when connecting to the EventSource API.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The default value is <see cref="Configuration.DefaultConnectionTimeout"/>.
+        /// </para>
+        /// <para>
+        /// Do not use this property if you are using <see cref="HttpClient"/> to specify a custom HTTP client. Doing so will cause <see cref="Build"/> to throw an exception.
+        /// </para>
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/ConfigurationBuilder.cs:74]: Added this text.

File: src/LaunchDarkly.EventSource/ConfigurationBuilder.cs [hunk 3]
```diff
@@ -138,101 +144,124 @@ public ConfigurationBuilder ReadTimeout(TimeSpan readTimeout)
...
-        {
-            this._messageHandler = handler;
-            return this;
-        }
-
-        /// <summary>
-        /// Sets the HTTP method that will be used when connecting to the EventSource API.
+        /// <remarks>
+        /// Do not use this property if you are using <see cref="HttpClient"/> to specify a custom HTTP client. Doing so will cause <see cref="Build"/> to throw an exception.
+        /// </remarks>
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/ConfigurationBuilder.cs:198]: Added this text to the contributed PR.

File: src/LaunchDarkly.EventSource/ConfigurationBuilder.cs [hunk 4]
```diff
@@ -138,101 +144,124 @@ public ConfigurationBuilder ReadTimeout(TimeSpan readTimeout)
...
+        /// <summary>
+        /// Specifies that EventSource should use a specific HttpClient instance for HTTP requests.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Normally, EventSource creates its own HttpClient and disposes of it when you dispose of the EventSource. If you provide your own HttpClient using this method, you are responsible for managing the HttpClient's lifecycle-- EventSource will not dispose of it.
+        /// </para>
+        /// <para>
+        /// EventSource will not modify this client's properties, so you should not the ConfigurationBuilder methods <see cref="MessageHandler"/> and <see cref="ConnectionTimeout"/> if you use this option. Doing so will cause <see cref="Build"/> to throw an exception.
+        /// </para>
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/ConfigurationBuilder.cs:216]: Rewote this doc comment from the contributed PR, for clarity & to match our style.

File: src/LaunchDarkly.EventSource/Resources.resx [hunk 5]
```diff
@@ -117,6 +117,12 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="Configuration_HttpClient_With_MessageHandler" xml:space="preserve">
+    <value>ConfigurationBuilder.HttpClient cannot be used with ConfigurationBuilder.MessageHandler.</value>
+  </data>
+  <data name="Configuration_HttpClient_With_ConnectionTimeout" xml:space="preserve">
+    <value>ConfigurationBuilder.HttpClient cannot be used with ConfigurationBuilder.ConnectionTimeout. Use Timeout property of HttpClient instead.</value>
+  </data>
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/Resources.resx:125]: Slight copyedits to this text from the contributed PR.

File: src/LaunchDarkly.EventSource/Configuration.cs [hunk 6]
```diff
@@ -7,18 +7,19 @@
 namespace LaunchDarkly.EventSource
 {
     /// <summary>
-    /// A class used 
+    /// An immutable class containing configuration properties for <see cref="EventSource"/>.
```
Comment (eli-darkly) [inline src/LaunchDarkly.EventSource/Configuration.cs:10]: Unrelated to the feature PR, and a bit embarrassing: this doc comment has apparently said nothing but "A class used" for the last 3 years.

**Review by bwoskow-ld — APPROVED** [review]
👍

**Review by gwhelanLD — APPROVED** [review]

**Pull request merged**